site/content/en/docs/Installation/upgrading.md

```diff
@@ -128,7 +128,74 @@ upgrades.
    between the Agones controller being recreated and GameServers being deleted doesn't occur, and GameServers can end up stuck in erroneous states.
 1. Start and complete your control plane upgrade(s).
 1. Start and complete your node upgrades.
-1. Scale your Fleets back up and/or recreate your GameServers. 
+1. Scale your Fleets back up and/or recreate your GameServers.
 1. Run any other tests to ensure the Agones installation is still working as expected.
 1. Close your maintenance window.
 7. Congratulations - you have now upgraded to a new version of Kubernetes! 👍
+
+{{% feature publishVersion="1.41.0" %}}
+## SDK Proto Compatibility Guarantees as of Agones v1.41.0
```

Comment:
This reason for "proto" is specifically SDK proto / SDK Server and does not apply to the client. The SDK client could be well out of date. For instance the C++, Rust, Unity, and Unreal clients don't have CountsAndLists at all even though it's going into Beta, and if no one updates them they may not have those features in GA. So it should be clear to the reader that this guarantee does not apply to the SDK clients. (Also the reason for using "proto" instead of "API" which is a bit more ambiguous.) If we were guaranteeing SDK clients we would need to have a separate discussion about deprecating clients if no one updates them regularly.

Reply:
But as an end user I don't care about the proto - I want to know if I am using my SDK client, will that stay backward compatible.

Much of this is covered in this section of the design doc: https://github.com/googleforgames/agones/issues/3766#client-sdks - which for this documentation is the more relevant part.

> The SDK client could be well out of date.

Why would that mater?

> or instance the C++, Rust, Unity, and Unreal clients don't have CountsAndLists at all even though it's going into Beta, and if no one updates them they may not have those features in GA.

Not an issue for backward or forward compatability.

> If we were guaranteeing SDK clients we would need to have a separate discussion about deprecating clients if no one updates them regularly.

I'm not sure why we would do that? Assuming the proto doesn't change (which it shouldn't for stable APIs, they should just work forever -- but the linked above RFC covers this.

Reply:
> For instance if the client has an API in Beta and no one moves it to GA, the Beta will break after 5 releases, so anyone using that client can't move past the version when it's removed from Beta

I don't think it will. There implementation doesn't go away - it's still there, it will all still work.

> Beta Client SDK APIs should be considered generally stable, but we may alter the parameters/returns if we need to

Beta APIs _may_ change, but that's also the risk of using Beta APIs - but are definitely considered more stable than Alpha.

 > We foresee that updating Client SDKs in game server binaries will be yearly toil, if using Stable APIs, or semiannual toil, if using any Beta API; this toil can be as simple as verifying that there were no deprecations in the period involved. Given how stable our core APIs have been, it may be possible to go multiple years without updating the Client SDK.

I am seeing some more room here to flesh out exactly what is guaranteed with a Beta SDK though - I think there's some ambiguity here that could be more concrete for sure.